### DISCUSSION TITLE
Is it possible to use XLNET as the pre-trained model for BERTopic?

### DISCUSSION THREAD

**Discussion by MarkWClements** [MaartenGr/BERTopic#992] [Q&A]
Hello,

I am wondering if it is possible to use a pre-trained XLNET model to get embeddings to pass to BERTopic via Flair. The reasoning is that I have a corpus of documents that are longer than the token limit for BERT (512). So rather than split each document into chunks, I was curious about passing the entire document to XLNET since it is designed to have no sequence length limit. Would something like the following work:

```
from bertopic import BERTopic
from flair.embeddings import TransformerDocumentEmbeddings

xlnet = TransformerDocumentEmbeddings("xlnet-base-cased")
embeddings = xlnet.embed(doc)

topic_model = BERTopic()
topics, probs = topic_model.fit_transform(docs, embeddings)

```

XLNET is a transformer model but is not a sentence-transformer model so I am curious if it will work well/be compatible with BERTopic.

**Comment by MaartenGr** [comment]
I have not used that model myself but there is definitely no harm in trying! Do note though that it might be worthwhile to apply [`.approximate_distribution`](https://maartengr.github.io/BERTopic/getting_started/distribution/distribution.html) after the initial fitting process as some documents might contain more than a single topic.

**Reply by MarkWClements** [reply]
Thank you for your response. Could you elaborate with more detail on why it might be worthwhile to apply ``.approximate_distribtuion`` specifically in the case of using XLNET for the embedding model along with some pseudo code?

**Reply by MaartenGr** [reply]
When you expect your documents to have more than 1 topic, it is typically advised to either split the documents or use `.approximate_distribution`. The reason for this is that BERTopic is essentially a clustering task from which the topics are extracted. So if you have very long documents and they are assigned to a single cluster, that might give issues. This can be resolved with `.approximate_distribution` since it allows for calculating the distribution of topics in long documents after the initial model has been trained. You can find more about the underlying algorithm of BERTopic [here](https://maartengr.github.io/BERTopic/algorithm/algorithm.html). I would also advise reading through the approximate distribution documentation [here](https://maartengr.github.io/BERTopic/getting_started/distribution/distribution.html).
